charms/slurmd/src/charm.py

```diff
@@ -90,6 +90,7 @@ def _on_install(self, event: InstallEvent) -> None:
 
             self.unit.status = MaintenanceStatus("Installing nhc")
             nhc.install()
+            nhc.generate_config()
```

Comment:
How would you feel about implicitly calling this function from `nhc.install()`? Since generating a base configuration is generally a part of the install process, we could just call `generate_config()` from within the install function. That way, as the install hook evolves over time, we don't have to worry about accidentally deleting this important function call from the hook.

Let me know your thoughts here!